src/doc/src/faq.md

```diff
@@ -259,3 +259,50 @@ Some issues we've seen historically which can cause crates to get rebuilt are:
 If after trying to debug your issue, however, you're still running into problems
 then feel free to [open an
 issue](https://github.com/rust-lang/cargo/issues/new)!
+
+### What does “version's conflict” mean and how to resolve it?
+
+> failed to select a version for `x` which could resolve this conflict
+
+Have you seen the error message above?
+
+This is one of the most annoying error message for Cargo users. There are several 
+situations may lead us to a version conflict. Below we'll walk through possible 
+causes and provide diagnostic techniques to help you out there:
+
+- The project and its dependencies use [links][links] to repeatedly link the local 
+library. Cargo forbids linking two packages with the same native library, so even 
+with multiple layers of dependencies it is not allowed. In this case, the error 
+message will prompt: `Only one package in the dependency graph may specify the same
+ links value`, you need to manually check and delete duplicate link values. however, 
+ that there are [conventions in place][conventions-in-place] to alleviate this.
+
+- When depending on different crates in the project, if these crates use the same 
+dependent library, but the version used is restricted, making it impossible to 
+determine the correct version, it will also cause conflicts.The error message will 
+prompt: `all possible versions conflict with previously selected packages`. Please 
+modify the version requirements to make them consistent.
+
+- If there are multiple versions of dependencies in the project, when using 
+[`direct-minimal-versions`][direct-minimal-versions], the minimum version 
+requirements cannot be met, which will cause conflicts. Please check in the 
+dependency graph of the project, the version requirements for dependencies 
+are inconsistent. Please modify the version requirements to make them consistent.
+
+- If the dependent crate does not have the features you choose, it will also 
+cause conflicts. At this time, you need to check the dependent version and its 
+features.
+
+Conflicts occur when merging branches or PRs, if there are non-trivial conflicts, 
```

Comment:
I think this deserves a list item, no? It looks like just one of those situations to me. Something like

```suggestion
- Conflicts may occur when merging branches or PRs, if there are non-trivial conflicts, 
```